pcdsdevices/signal.py

```diff
@@ -794,97 +802,80 @@ class AvgSignal(Signal):
 
     Parameters
     ----------
-    signal : Signal
+    signal : Signal or str
         Any subclass of `ophyd.signal.Signal` that returns a numeric value.
         This signal will be subscribed to be `AvgSignal` to calculate the mean.
+        Parent classes can pass a str instead that matches the attr name
+        of one of their component signals.
 
     averages : int
-        The number of `SUB_VALUE` updates to include in the average. New values
+        The number of ``SUB_VALUE`` updates to include in the average. New values
         after this number is reached will begin overriding old values.
+
+    duration : float, optional
+        The number of seconds to wait before returning trigger complete. Nominally this
+        should be set to averages divided by the expected update rate of the signal.
+        If omitted, we will not reset the buffer or wait for values at scan points.
     """
 
-    def __init__(self, signal, averages, *, name, parent=None, **kwargs):
+    def __init__(
+        self,
+        signal: Signal | str,
+        averages: int,
+        duration: float | None = None,
+        *,
+        name: str,
+        parent: Device | None = None,
+        **kwargs,
+    ):
         super().__init__(name=name, parent=parent, **kwargs)
         if isinstance(signal, str):
             signal = getattr(parent, signal)
         self.raw_sig = signal
         self._lock = RLock()
         self.averages = averages
+        self.duration = duration
         self.raw_sig.subscribe(self._update_avg)
 
     @property
-    def connected(self):
+    def connected(self) -> bool:
         return self.raw_sig.connected
 
     @property
-    def averages(self):
+    def averages(self) -> int:
```

Comment:
I dislike this name but I dislike breaking people's functionality more

Reply:
In retrospect, I also really dislike this old name!